### ISSUE TITLE
AttributeError: 'ProgrammingError' object has no attribute 'code'

### ISSUE THREAD

**Issue by RalfJung** [opatut/dudel#77]
I get the following error when a request is made to the root of the dudel WSGI application:
```
[Sat Mar 21 14:26:10 2015] [error] [client 91.6.237.0] mod_wsgi (pid=29081): Exception occurred processing WSGI script '$HOME/dudel/dudel.wsgi'.
[Sat Mar 21 14:26:10 2015] [error] [client 91.6.237.0] Traceback (most recent call last):
[Sat Mar 21 14:26:10 2015] [error] [client 91.6.237.0]   File "$HOME/.local/lib/python2.7/site-packages/raven/middleware.py", line 35, in __call__
[Sat Mar 21 14:26:10 2015] [error] [client 91.6.237.0]     iterable = self.application(environ, start_response)
[Sat Mar 21 14:26:10 2015] [error] [client 91.6.237.0]   File "/usr/lib/python2.7/dist-packages/flask/app.py", line 1506, in wsgi_app
[Sat Mar 21 14:26:10 2015] [error] [client 91.6.237.0]     response = self.make_response(self.handle_exception(e))
[Sat Mar 21 14:26:10 2015] [error] [client 91.6.237.0]   File "/usr/lib/python2.7/dist-packages/flask/app.py", line 1197, in handle_exception
[Sat Mar 21 14:26:10 2015] [error] [client 91.6.237.0]     return handler(e)
[Sat Mar 21 14:26:10 2015] [error] [client 91.6.237.0]   File "$HOME/dudel/dudel/views.py", line 1034, in error
[Sat Mar 21 14:26:10 2015] [error] [client 91.6.237.0]     return render_template("error.html", error=err), err.code
[Sat Mar 21 14:26:10 2015] [error] [client 91.6.237.0] AttributeError: 'ProgrammingError' object has no attribute 'code'
```

I am fairly clueless what could be the cause of this error.

**Comment by RalfJung** [comment]
After making sure I don't use the system version of `flask`, the error persists:
```
[Sat Mar 21 16:53:43 2015] [error] Exception TypeError: "'NoneType' object is not callable" in <bound method Popen.__del__ of <subprocess.Popen object at 0xb96de12c>> ignored
[Sat Mar 21 16:53:45 2015] [error] Raven is not configured (logging is disabled). Please see the documentation for more information.
[Sat Mar 21 16:53:46 2015] [error] [client 91.6.237.0] mod_wsgi (pid=5931): Exception occurred processing WSGI script '$HOME/dudel/dudel.wsgi'.
[Sat Mar 21 16:53:46 2015] [error] [client 91.6.237.0] Traceback (most recent call last):
[Sat Mar 21 16:53:46 2015] [error] [client 91.6.237.0]   File "$HOME/.local/lib/python2.7/site-packages/raven/middleware.py", line 35, in __call__
[Sat Mar 21 16:53:46 2015] [error] [client 91.6.237.0]     iterable = self.application(environ, start_response)
[Sat Mar 21 16:53:46 2015] [error] [client 91.6.237.0]   File "$HOME/.local/lib/python2.7/site-packages/flask/app.py", line 1820, in wsgi_app
[Sat Mar 21 16:53:46 2015] [error] [client 91.6.237.0]     response = self.make_response(self.handle_exception(e))
[Sat Mar 21 16:53:46 2015] [error] [client 91.6.237.0]   File "$HOME/.local/lib/python2.7/site-packages/flask/app.py", line 1410, in handle_exception
[Sat Mar 21 16:53:46 2015] [error] [client 91.6.237.0]     return handler(e)
[Sat Mar 21 16:53:46 2015] [error] [client 91.6.237.0]   File "$HOME/dudel/dudel/views.py", line 1034, in error
[Sat Mar 21 16:53:46 2015] [error] [client 91.6.237.0]     return render_template("error.html", error=err), err.code
[Sat Mar 21 16:53:46 2015] [error] [client 91.6.237.0] AttributeError: 'ProgrammingError' object has no attribute 'code'
```

**Comment by RalfJung** [comment]
Here's what `str(err)` of the error that it tries to display looks like:
```
(_mysql_exceptions.ProgrammingError) (1146, "Table 'dudel.poll' doesn't exist") [SQL: u'SELECT poll.id AS poll_id, poll.title AS poll_title, poll.description AS poll_description, poll.slug AS poll_slug, poll.type AS poll_type, poll.created AS poll_created, poll.owner_id AS poll_owner_id, poll.due_date AS poll_due_date, poll.anonymous_allowed AS poll_anonymous_allowed, poll.public_listing AS poll_public_listing, poll.require_login AS poll_require_login, poll.require_invitation AS poll_require_invitation, poll.show_results AS poll_show_results, poll.send_mail AS poll_send_mail, poll.one_vote_per_user AS poll_one_vote_per_user, poll.allow_comments AS poll_allow_comments, poll.deleted AS poll_deleted, poll.show_invitations AS poll_show_invitations, poll.timezone_name AS poll_timezone_name \\nFROM poll \\nWHERE poll.deleted = false AND poll.public_listing = true AND poll.due_date >= %s ORDER BY poll.created DESC'] [parameters: (datetime.datetime(2015, 3, 21, 15, 59, 10, 544633),)]
```

Indeed I forgot to do "make init" *oops*. But nevertheless, it should not try to fetch the error code from an exception that doesn't have one.

**Comment by opatut** [comment]
You are right. I had no idea the exception handler tried to catch *every* exception. Gonna fix.

**Comment by RalfJung** [comment]
I think this has actually already been fixed by 2abc219bf87b10adc27744ff29b06ff8b4f0a321.

**Comment by Akasch** [comment]
yes this schould be fixed with 2abc219 .
